spring-boot-modules/spring-boot-testing/src/main/java/com/baeldung/boot/configurationproperties/CustomCredentialsConverter.java

```diff
@@ -0,0 +1,16 @@
+package com.baeldung.boot.configurationproperties;
+
+import org.springframework.boot.context.properties.ConfigurationPropertiesBinding;
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.stereotype.Component;
+
+@Component
+@ConfigurationPropertiesBinding
+public class CustomCredentialsConverter implements Converter<String, Credentials> {
+
+	@Override
+	public Credentials convert(String source) {
+		String data[] = source.split(",");
```

Comment:
Typically we would put Array designators "[]" on the type, not the variable.

Reply:
Fixed.